src/sentry/api_gateway/proxy.py

```diff
@@ -30,21 +34,30 @@ def stream_response():  # type: ignore
     )
     # Add Headers to response
     for header, value in response.headers.items():
-        streamed_response[header] = value
+        if not is_hop_by_hop(header):
+            streamed_response[header] = value
+
     streamed_response["X-Sentry-Proxy-URL"] = remote_url
     return streamed_response
 
 
+def _strip_request_headers(headers) -> dict:
+    header_dict = {}
+    for header, value in headers.items():
+        if header not in HEADER_STRIKE_LIST:
+            header_dict[header] = value
+
+
 def proxy_request(request: Request, org_slug: str) -> StreamingHttpResponse:
     """Take a django request object and proxy it to a remote location given an org_slug"""
     from sentry.types.region import get_region_for_organization
 
     target_url = get_region_for_organization(None).to_url(request.path)
-
+    header_dict = _strip_request_headers(request.headers)
```

Comment:
A test case for the `Content-Length` header being stripped would be nice to have 👀